stan/math/prim/prob/hmm_latent_rng.hpp

```diff
@@ -70,18 +70,22 @@ inline std::vector<int> hmm_latent_rng(const T_omega& log_omegas,
   Eigen::Map<Eigen::VectorXd> probs_vec(probs.data(), n_states);
   probs_vec = alphas.col(n_transitions) / alphas.col(n_transitions).sum();
   boost::random::discrete_distribution<> cat_hidden(probs);
-  hidden_states[n_transitions] = cat_hidden(rng);
+  hidden_states[n_transitions] = cat_hidden(rng) + 1;
```

Comment:
@charlesm93, there's a terribly named variable that stores the indexing. If you wanted to be proper about this, you could change this (and other instances) of `+ 1` with `+ stan::error_index::value`. https://github.com/stan-dev/math/blob/develop/stan/math/prim/meta/error_index.hpp

It's not only terribly named... it has no doc and I don't know how you would have known about it. I'll create issues to tackle that later.

The intention was if someone really wanted to 0-index things (like if it were from a C++ function), they'd compile with `-DERROR_INDEX=0` and that would magically make things like this, bounds checks on arrays, and other places where there is indexing, be 0-indexed. By default, it's 1-indexed.

Alternately, we can just ditch the generality and explicitly say that everything is 1-indexed, as confusing as it is with C++ being 0-indexed, and make life simpler.

Anyway, just adding a comment so you know that there's a variable that defines the indexing. If you want to switch it, great. If not, also great.

Reply:
@charlesm93 I think it makes sense to do what @syclik is saying here.

I think how this works is you just add/subtract `stan::error_index::value` everywhere before you added/subtracted `1` (including the tests). It's all tested in 1-indexing mode (how Stan uses it) and then if someone wants the 0-indexing they can do the make flag.

Release is the 19th and code freeze is the 12th for reference.

Reply:
@charlesm93 since this was sitting here and it was a relatively simple but confusing change I just did it (and made Bob's edit). Have a look and if you approve we can merge this.

Reply:
@bbbales2 Thanks for jumping in. I agree with what @syclik proposes. I'll take a quick look.

Reply:
@bbbales2 you missed one +1, I translated it into an index error. Once the integration tests pass, I'll merge it (small caprice, I love pushing the merge button!)